spec/controllers/projects_controller_spec.rb

```diff
@@ -1,386 +1,350 @@
-#           end
-#         end
-#       end
-#     end
-#   end
-# end
+require 'rails_helper'
+
+RSpec.describe ProjectsController, type: :controller do
+  let(:student) { double('Student', id: '1', admin?: false) }
+  let(:admin) { double('Admin', id: '2', admin?: true) }
+  let(:course) { double('Course', id: '1') }
+  let(:lesson) {
+    double('Lesson', id: '1', slug: 'building-blocks', course: course)
+  }
+
+  let(:valid_project_attrs) { { repo_url: 'https://github.com/user/repo' } }
+  let(:invalid_project_attrs) { { repo_url: 'https://github.co' } }
+
+  let(:valid_params) {
+    {
+      lesson_id: lesson.slug,
+      project: valid_project_attrs
+    }
+  }
+
+  let(:invalid_params) {
+    {
+      lesson_id: lesson.slug,
+      project: invalid_project_attrs
+    }
+  }
+
+  let(:projects) { double('ProjectsRelation') }
+  let(:projects_association) { double('ProjectsAssociation') }
+  let(:student_project) {
+    double('Project', valid_project_attrs.merge(id: '1', user: student))
+  }
+
+  let(:admin_project) {
+    double('Project', valid_project_attrs.merge(id: '2', user: admin))
+  }
+
+  describe 'GET #index' do
+    before do
+      stub_lesson
+      allow(Project).to receive(:all_submissions).with('1').and_return(projects)
+      allow(projects).to receive(:order).with(updated_at: :desc).and_return(projects)
+      allow(projects).to receive(:page).and_return(projects)
+      allow(projects).to receive(:per).with(50).and_return(projects)
+    end
+
+    it 'renders the project index template' do
+      get :index, params: { lesson_id: 'building-blocks' }
+      expect(response).to render_template(:index)
+    end
+
+    it 'assigns @projects' do
+      get :index, params: { lesson_id: 'building-blocks' }
+      expect(assigns(:projects)).to eq(projects)
+    end
+  end
+
+  describe 'POST #create' do
+    subject { post :create, params: valid_params, xhr: true }
+
+    context 'unauthenticated' do
+      it_behaves_like 'unauthenticated request'
+    end
+
+    context 'authenticated' do
+      before do
+        stub_current_user_with(student)
+        stub_lesson
+        allow(controller).to receive(:new_project).and_return(student_project)
+      end
+
+      context 'valid values' do
+        it 'assigns recent submissions' do
+          allow(student_project).to receive(:save).and_return(true)
+          allow(controller).to receive(:latest_projects).and_return(projects)
+          subject
+          expect(assigns(:projects)).to eq(projects)
+        end
+
+        it 'renders the create template' do
+          allow(student_project).to receive(:save).and_return(true)
+          allow(controller).to receive(:latest_projects).and_return(projects)
+          expect(subject).to render_template(:create)
+        end
+      end
+
+      context 'invalid values' do
+        subject { post :create, params: invalid_params, xhr: true }
+
+        it 'renders the create template' do
```

Comment:
This  functions pretty much the same as when valid params are passed in so we can remove this test. Invalid params should be tested in end to end testing imo.